tools/state-viewer/src/cli.rs

```diff
@@ -679,20 +718,79 @@ pub struct ViewTrieCmd {
     /// For format=pretty this measures depth in terms of key nibbles.
     #[clap(long)]
     max_depth: u32,
+    /// Limits how many entries are printed to the output.
+    #[clap(long)]
+    limit: Option<u32>,
+    /// Filters output to only show records of given type.
+    #[clap(long)]
+    rtype: Option<StateRecordType>,
+    /// Will only print nodes with key being a prefix of or lexicographically greater than `from`.
+    /// Only works for the `full` format. The argument should be wrapped in parentheses using mixed encoding.
+    #[clap(long)]
+    from: Option<String>,
+    /// Will only print nodes with key being lexicographically less than or equal to `to`.
+    /// Only works for the `full` format. The argument should be wrapped in parentheses using mixed encoding.
+    #[clap(long)]
+    to: Option<String>,
 }
 
 impl ViewTrieCmd {
     pub fn run(self, store: Store) {
         let hash = CryptoHash::from_str(&self.hash).unwrap();
+        let type_str: Option<&str> = self.rtype.map(|s| s.into());
+        let from_vec = self.from.as_ref().map(|s| {
+            Self::decode_mixed_hex_to_nibbles(s).expect("Failed to decode `from` argument.")
+        });
+        let to_vec = self.to.as_ref().map(|s| {
+            Self::decode_mixed_hex_to_nibbles(s).expect("Failed to decode `to` argument.")
+        });
```

Comment:
Oh my, why hex? Having human readable acccoun ids is arguably one of the better properties of the near blockchain ;)